docs/docs/spec.md

```diff
@@ -938,17 +928,27 @@ the output of the following operation:
     > in all worlds. Now consider a null-unmarked class that declares a method
     > `Comparable<Foo> foo()`, which by JSpecify rules has a type argument `Foo
     > UNSPECIFIED`. In this example, the question is what type
-    > `foo().compare(...)` accepts, and the answer changes as a result of this
-    > part of the subsitution rules:
+    > `foo().compare(...)` accepts. That question demonstrates the effect of
+    > this part of the subsitution rule:
     >
-    > -   If JSpecify were to directly subsitute `Foo UNSPECIFIED` in for `T`,
-    >     then the parameter type, which started out as non-null, would become
-    >     unspecified during subsitution. As a result, lenient checkers would
-    >     allow the call `foo().compare(null)`, since `Foo UNSPECIFIED` is
+    > -   Without this part of the rule, JSpecify would directly subsitute `Foo
+    >     UNSPECIFIED` in for `T`. Then the parameter type, which started out as
```

Comment:
Do we need the "in" here? "substituted X in for Y" vs "substituted X for Y".

_This is the most consequential comment I've ever made._

Reply:
Done.

(I started wondering whether we should say "for `T NO_CHANGE`" or otherwise attempt to be more precise, but then I stuffed it back down.)